senders/test_server_test.go

```diff
@@ -0,0 +1,52 @@
+package senders
+
+import (
+	"bufio"
+	"compress/gzip"
+	"net/http"
+	"net/http/httptest"
+)
+
+func startTestServer() *testServer {
+	handler := &testServer{}
+	server := httptest.NewServer(handler)
+	handler.httpServer = server
+	handler.URL = server.URL
+	return handler
+
+}
+
+type testServer struct {
+	MetricLines []string
+	httpServer  *httptest.Server
+	URL         string
+}
+
+func (s *testServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	newLines, err := decodeMetricLines(request)
+	if err != nil {
+		writer.WriteHeader(500)
+	}
+	for _, metricLine := range newLines {
```

Comment:
You can just say ```s.MetricLines = append(s.MetricLines, newLines...)```

Reply:
ah! I thought there was a splat syntax but tried `...newLines` and when that didn't work I just rolled with the for loop 🤣